File: 백준/Silver/11659. 구간 합 구하기 4/구간 합 구하기 4.java

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int howManyNumber = Integer.parseInt(st.nextToken());
        int howManyQuery = Integer.parseInt(st.nextToken());

        st = new StringTokenizer(br.readLine());
        int[] windowSum = new int[howManyNumber + 1];
        for (int i = 0; i < howManyNumber; i++) {
            windowSum[i + 1] = windowSum[i] + Integer.parseInt(st.nextToken());
        }

        for (int i = 0; i < howManyQuery; i++) {
            st = new StringTokenizer(br.readLine());
            int queryI = Integer.parseInt(st.nextToken());
            int queryJ = Integer.parseInt(st.nextToken());
            System.out.println(windowSum[queryJ] - windowSum[queryI - 1]);
        }
    }
}

File: 백준/Silver/11659. 구간 합 구하기 4/구간 합 구하기 4.kt
import java.io.*
import java.util.StringTokenizer

fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {
    
    val insert1 = StringTokenizer(readLine())
    val N = insert1.nextToken().toInt()
    val M = insert1.nextToken().toInt()

    val insert2 = StringTokenizer(readLine())
    // 구간합 계산
    val sumOfIntervals = IntArray(N + 1)
    repeat(N) { i ->
        sumOfIntervals[i+1] = sumOfIntervals[i] + insert2.nextToken().toInt()
    }

    val stringBuilder = StringBuilder()
    // 구간합의 구간 입력과 바로 계산
    repeat(M) {
        val insert3 = StringTokenizer(readLine())
        val startIndex = insert3.nextToken().toInt()
        val endIndex = insert3.nextToken().toInt()
        val result = sumOfIntervals[endIndex] - sumOfIntervals[startIndex - 1]
        stringBuilder.appendLine(result)
    }
    
    print(stringBuilder.toString())
}